File: Orimath.ViewPlugins/Controls/ResettableObservableCollection.cs
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Collections.ObjectModel;
using System.ComponentModel;

namespace Orimath.Controls
{
    public class ResettableObservableCollection<T> : ObservableCollection<T>
    {
        public ResettableObservableCollection() : base() { }

        public ResettableObservableCollection(IEnumerable<T> collection) : base(collection) { }

        public ResettableObservableCollection(List<T> list) : base(list) { }

        public void Reset(IEnumerable<T> newItems)
        {
            Items.Clear();
            ((List<T>)Items).AddRange(newItems);
            OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
            OnPropertyChanged(new PropertyChangedEventArgs(nameof(Count)));
        }
    }
}

File: Orimath.ViewPlugins/Controls/ResettableObservableCollection.fs
namespace Orimath.Controls
open System.Collections.Generic
open System.Collections.Specialized
open System.Collections.ObjectModel
open System.ComponentModel

type ResettableObservableCollection<'T>(collection: seq<'T>) =
    inherit ObservableCollection<'T>(collection)

    member this.Reset(newItems) =
        this.Items.Clear()
        (this.Items :?> List<'T>).AddRange(newItems)
        this.OnCollectionChanged(NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset))
        this.OnPropertyChanged(PropertyChangedEventArgs(nameof this.Count))

    new() = ResettableObservableCollection(array.Empty())
